Remove bogus early return from `nth_percentile_fetch_statistics`

KrisShannon opened the pull request:
If there are more than one data source then the early return when a single
data source is missing data will cause all data sources to be ignored,  even
those that do have data.

Fixes #4263

netniV reviewed (changes requested):
PLease can you add the CHANGELOG entry for the issue that this fixes.

netniV commented:
I've requested a review on this by @TheWitness as he changed it in #3340 as part of commit 621091f7a.  I think the fix seems OK, just want to be sure as I suspect this was added to speed things up.

KrisShannon commented:
Should this PR actually be against [1.2.x](https://github.com/Cacti/cacti/tree/1.2.x)?

It's a simple fix for a bug in the 1.2 branch.

netniV commented:
Technically, yes it should be since it's not a 1.3 specific fix.  Like I said the fix seems straight forward but just wondering if there was a logic behind it that I'm not seeing myself.

KrisShannon committed:
Remove bogus early return from `nth_percentile_fetch_statistics`

If there are more than one data source then the early return when a single
data source is missing data will cause all data sources to be ignored,  even
those that do have data.

KrisShannon commented:
CHANGELOG entry added
